pkg/cmd/cluster/connect.go

```diff
@@ -445,11 +445,11 @@ func getUserAndPassword(clusterDef *appsv1alpha1.ClusterDefinition, compDef *app
 	}
 	// 0.8 API connect
 	if secret == nil && compDef != nil {
-		for i := range compDef.Spec.SystemAccounts {
-			if compDef.Spec.SystemAccounts[i].InitAccount {
-				for j, s := range secrets.Items {
-					if strings.Contains(s.Name, compDef.Spec.SystemAccounts[j].Name) {
-						secret = &secrets.Items[j]
+		for _, account := range compDef.Spec.SystemAccounts {
+			if account.InitAccount {
+				for _, s := range secrets.Items {
+					if s.Name == fmt.Sprintf("%s-%s-account-%s", cluster, compDef.Name, account.Name) {
+						secret = &s
```

Comment:
DO NOT return the `s` address that is a temp variable used by for loop, use index to fetch the ith item poiint and return.